### PR TITLE
Extending Workbox blog post

### LINKED ISSUE #3964: content: Extending Workbox: Custom Strategies & Plugins [2020-10-22]
* [thread](https://groups.google.com/a/google.com/g/webdev-content/c/Dq06nXsnVWo)
* [draft](https://docs.google.com/presentation/d/1fnp_nAuMkuP_h2QjtxJ7Lg8TT-DHT3V7l9oQzHiyOMc/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by jeffposnick**
Fixes #3964

This content was previously reviewed by @kaycebasques.

Changes proposed in this pull request:

- Blog post to accompany my CDS talk. 
- Currently marked as `draft: true` ~~but needs to go live prior to my talk premiere.~~ (in terms of timing, it's okay if the URL works but the article isn't "public")

### CODE REVIEW

File: src/site/content/en/blog/extending-workbox/index.md
Comment (jeffposnick) [inline src/site/content/en/blog/extending-workbox/index.md:12]: Are there any other tags to use for the CDS-related blog posts?

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:7]:
```suggestion
  Write your own reusable, shareable strategies and plugins.
authors: jeffposnick
date: 2020-12-10
```
Let's attribute this post to you, yes?

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [reply src/site/content/en/blog/extending-workbox/index.md:12]: There's no CDS-specific posts but I'll check if there are other general tags that are relevant here: https://github.com/GoogleChrome/web.dev/blob/master/src/site/_data/tagsData.json

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:15]:
```suggestion
  - service-worker
  - offline
  - network
  - progressive-web-apps
  - performance
---
```

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:55]:
```suggestion
### Handle simultaneous, duplicate requests with DedupeNetworkFirst
```
Didn't catch this during the early review: Let's update these sections to follow the "X with Y" pattern, where X is the general task that's solved and Y is the name of the feature?

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:109]:
```suggestion
### Create a race between the cache and network with CacheNetworkRace
```
Not sure how to summarize this section… this suggestion is probably good enough (unless you've got something better)

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:113]: In practice would the network ever be faster here? (just curious; the thought just came to me; no action needed)

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:148]:
```suggestion
## StategyHandler: the recommended approach for creating custom strategies
```
`Using the StrategyHandler` doesn't give me a clue about why I would want to use it

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:227]:
```suggestion
## Custom plugins
```
Nitpick: sentence case is our site-wide convention

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:235]:
```suggestion
### Lifecycle event overview
```

File: src/site/content/en/blog/extending-workbox/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/extending-workbox/index.md:310]:
```suggestion
### Lifecycle events example: FallbackOnErrorPlugin
```

**Review by kaycebasques — COMMENTED** [review]
@jeffposnick thank you for the tidy PR. Please also:

- [ ] reference this page from web.dev/reliable (edit `/src/site/_data/paths/reliable.json`)

**Review by kaycebasques — CHANGES_REQUESTED** [review]
whoops, meant to set the last review as "changes requested"

File: src/site/content/en/blog/extending-workbox/index.md
Comment (jeffposnick) [reply src/site/content/en/blog/extending-workbox/index.md:118]: Not usually, but there are some, e.g., old Android devices with really bad flash storage that might be on a fast network.

**Comment by jeffposnick** [comment]
I've edited `/src/site/_data/paths/reliable.json` to add my article as [per your request](https://github.com/GoogleChrome/web.dev/pull/4296#pullrequestreview-538089954), but it's not showing up in the list at https://deploy-preview-4296--web-dev-staging.netlify.app/reliable/

Is that because my article is checked in to the file system under `blog` at `src/site/content/en/blog/extending-workbox/index.md` instead of under `reliable`, e.g. `src/site/content/en/reliable/extending-workbox/index.md`?

**Comment by jeffposnick** [comment]
Hi @kaycebasques—this is still waiting on your re-review/approval.

**Review by kaycebasques — APPROVED** [review]

**Pull request merged**
